### DISCUSSION TITLE
How to add a line between headers and table?

### DISCUSSION THREAD

**Discussion by vulns-are-features-too** [fdehau/tui-rs#458] [Q&A]
Following `/examples/table.rs`, I've made a simple table. I want a horizontal line between the headers and table, but that's now shown in the example.
 How can I add a horizontal line between the headers and table?

**Comment by c64zottel** [comment]
I had the same Problem: just a straight horizontal line. Took me a day, but I learned a lot about tui-rs.
Here are my two possible solutions:
- Print a string with Unicode symbol 2500:
 `terminal.current_buffer_mut().set_string(1, 7, "\u{2500}────────────────────", Style::default());`
- Draw a rectangle with height 1 and Borders::TOP only:
>     let line = Block::default().borders(Borders::TOP);
>     let rect = Rect { x: 0, y: 23, width: 80, height: 1 };
>     f.render_widget(line, rect);

There is also the possibility to use canvas, which has a struct Line: [](https://docs.rs/tui/0.10.0/tui/widgets/canvas/struct.Line.html)
But the default line is dotted. Maybe it's possible to change the painter or style.

I am happy with the Block-Line for now.

![image](https://user-images.githubusercontent.com/5398700/169476698-4eda72bf-e70c-49e2-83b2-9579324c82f1.png)
